storage/src/journal/fixed.rs

```diff
@@ -272,36 +280,45 @@ impl<B: Blob, E: Storage<B>, A: Array> Journal<B, E, A> {
             Some(blob) => blob,
             None => return Err(Error::MissingBlob(rewind_to_blob_index)),
         };
-        let rewind_to_offset = (rewind_to_size % self.cfg.items_per_blob) * Self::CHUNK_SIZE as u64;
         rewind_blob.truncate(rewind_to_offset).await?;
+
         Ok(())
     }
 
+    /// Return the position of the oldest item in the journal that remains readable.
+    ///
+    /// Note that this value could be older than the `min_item_pos` last passed to prune.
+    pub async fn oldest_retained_pos(&self) -> Result<Option<u64>, Error> {
+        if self.oldest_blob().1.len().await? == 0 {
+            return Ok(None);
+        }
+        Ok(Some(self.oldest_blob().0 * self.cfg.items_per_blob))
```

Comment:
nit: This is probably overkill but may be worth putting a comment that any retained blob always starts with an "item in the correct position" (or something like that).

Reply:
added a comment... does that address the concern?